pages/tree.tsx

```diff
@@ -1,13 +1,41 @@
 import type { NextPage } from "next";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
 import TreeNodeDetails from "../components/Tree/TreeNodeDetails/TreeNodeDetails";
 import TreeWithNavigation from "../components/Tree/TreeWithNavigation/TreeWithNavigation";
 import { TreeContextWrapper } from "../context/tree";
+import s from "../styles/TreePage.module.css";
 
-const TreePage: NextPage = () => (
-  <TreeContextWrapper>
-    <TreeWithNavigation />
-    <TreeNodeDetails />
-  </TreeContextWrapper>
-);
+const TreePage: NextPage = () => {
+  const router = useRouter();
+  const [rootId, setRootId] = useState<string>();
+
+  useEffect(() => {
```

Comment:
Весь useEffect можно заменить на 
```suggestion
  const queryRootId = root
      ? Array.isArray(root)
        ? root[0]
        : root
      : undefined;
```